sqlmesh/core/context.py

```diff
@@ -942,10 +942,15 @@ def config_for_path(self, path: Path) -> t.Tuple[Config, Path]:
 
     def config_for_node(self, node: str | Model | Audit) -> Config:
         if isinstance(node, str):
-            return self.config_for_path(self.get_snapshot(node, raise_if_missing=True).node._path)[
-                0
-            ]  # type: ignore
-        return self.config_for_path(node._path)[0]  # type: ignore
+            path = self.get_snapshot(node, raise_if_missing=True).node._path
+            if path is None:
+                return self.config
+            config = self.config_for_path(path)
+            return config[0]
+        path = node._path
+        if path is None:
+            return self.config
+        return self.config_for_path(path)[0]  # type: ignore
```

Comment:
Nit: I'd suggest consolidating these two branches into one. If it's a string, get the node and let it fall through.